src/ui/components/hooks/useCurrentTime.ts

```diff
@@ -0,0 +1,27 @@
+import { useCallback, useContext, useSyncExternalStore } from 'react';
+import { PlayerContext } from '@theoplayer/react-native-ui';
+import { type PlayerEventMap, PlayerEventType } from 'react-native-theoplayer';
+
+const TIME_CHANGE_EVENTS = [PlayerEventType.TIME_UPDATE, PlayerEventType.SEEKING, PlayerEventType.SEEKED] satisfies ReadonlyArray<
+  keyof PlayerEventMap
+>;
+
+/**
+ * Returns {@link react-native-theoplayer!THEOplayer.duration | the player's current time}, automatically updating whenever it changes.
+ *
+ * This hook must only be used in a component mounted inside a {@link THEOplayerDefaultUi} or {@link UiContainer},
+ * or alternatively any other component that provides a {@link PlayerContext}.
+ *
+ * @group Hooks
+ */
+export const useCurrentTime = () => {
+  const player = useContext(PlayerContext).player;
```

Comment:
Nit-pick:
```suggestion
  const { player } = useContext(PlayerContext);
```

Reply:
I also replaced this with destructuring assignment in the other hooks.